src/main/java/com/appdynamics/extensions/haproxy/HAProxyMonitorTask.java

```diff
@@ -0,0 +1,348 @@
+                    sheet.addCell(label);
+                }
+                j++;
+            }
+            workbook.write();
+            workbook.close();
+            outputStream.close();
+        } catch (Exception e) {
+            logger.error("Error while writing response to workbook stream ", e);
+            throw new RuntimeException("Error while writing response to workbook stream");
+        }
+    }
+
+    /**
+     * Returns all proxies and proxy types as a Map of row index and Proxy Name.
+     * Column Index is 0 for Px Name and 1 for Server Name
+     *
+     * @param columnIndex
+     * @return
+     */
+    private Map<Integer, String> getAllProxyAndTypes(int columnIndex) {
+        Map<Integer, String> proxiesMap = new HashMap<Integer, String>();
+        int rows = getSheet().getRows();
+        for (int i = 1; i < rows; i++) {
+            proxiesMap.put(i, getSheet().getCell(columnIndex, i).getContents());
+        }
+        return proxiesMap;
+    }
+
+    /**
+     * Returns the zeroth sheet of the workbook
+     *
+     * @return
+     */
+    private Sheet getSheet() {
+        return workbook.getSheet(0);
+    }
+
+
+    private Map<String, String> buildRequestMap(Map haServer) {
+        Map<String, String> requestMap = new HashMap<String, String>();
+        requestMap.put("host", (String) haServer.get("host"));
+        requestMap.put("port", String.valueOf(haServer.get("port")));
+        requestMap.put("useSsl", String.valueOf(haServer.get("useSsl")));
+        requestMap.put("username", (String) haServer.get("username"));
+        requestMap.put("password", (String) haServer.get("password"));
+        return requestMap;
+    }
+
+
+    private Map<Integer, String> filterProxies(Map<String, String> haServerArgs, Map<Integer, String> allProxies) {
+        List<String> proxiesListedInConfigFile = Lists.newArrayList();
+        if (haServerArgs.containsKey("proxynames") && !Strings.isNullOrEmpty(haServerArgs.get("proxynames"))) {
+            proxiesListedInConfigFile = Arrays.asList(haServerArgs.get("proxynames").split(","));
+        }
+        if (proxiesListedInConfigFile.size() != 0) {
+            for (Iterator<Map.Entry<Integer, String>> it = allProxies.entrySet().iterator(); it.hasNext(); ) {
+                Map.Entry<Integer, String> entry = it.next();
+                if (!proxiesListedInConfigFile.contains(entry.getValue())) {
+                    it.remove();
+                }
+            }
+        }
+        return allProxies;
+    }
+
+
+    private void printStats(Map<String, String> haServerArgs, Map<Integer, String> proxiesToBeMonitored, Map<Integer, String> proxyTypes) {
+
+        List<String> excludedStats = new ArrayList<String>();
+        if (haServerArgs.containsKey("excludeStats") && !Strings.isNullOrEmpty(haServerArgs.get("excludeStats"))) {
+            excludedStats = Arrays.asList(haServerArgs.get("excludeStats").split(","));
+        }
```

Comment:
we can remove the concept of excludeStats from config.yml, with metrics.xml in place excluding/including metrics can be take care from metrics.xml itself